From 8bc732e8b03ebb4309f4b42626917148d176db49 Mon Sep 17 00:00:00 2001
From: Jeremy Harris <jgh146exb@wizmail.org>
Date: Sun, 5 Oct 2014 21:31:20 +0100
Subject: [PATCH] Remove limit on remove_headers item size. Bug 1533

---
 doc/doc-txt/ChangeLog | 5 +++++
 src/src/transport.c   | 3 +--
 2 files changed, 6 insertions(+), 2 deletions(-)

| diff --git a/doc/doc-txt/ChangeLog b/doc/doc-txt/ChangeLog
| index 76ecc20..0b03894 100644
| --- a/doc/doc-txt/ChangeLog
| +++ b/doc/doc-txt/ChangeLog
| @@ -44,6 +44,11 @@ JH/05 Fix results-pipe from transport process.  Several recipients, combined
|        to notice due to the introduction of conection certificate information,
|        the item size being so much larger.  Found and fixed by Wolfgang Breyha.
|  
| +JH/06 Bug 1533: Fix truncation of items in headers_remove lists.  A fixed
| +      size buffer was used, resulting in syntax errors when an expansion
| +      exceeded it.
| +
| +
|  Exim version 4.84
|  -----------------
|  TL/01 Bugzilla 1506: Re-add a 'return NULL' to silence complaints from static
diff --git a/src/transport.c b/src/transport.c
index 31437b1..15c30bf 100644
--- a/src/transport.c
+++ b/src/transport.c
@@ -643,8 +643,7 @@ for (h = header_list; h != NULL; h = h->next) if (h->type != htype_old)
       {
       int sep = ':';         /* This is specified as a colon-separated list */
       uschar *s, *ss;
-      uschar buffer[128];
-      while ((s = string_nextinlist(&list, &sep, buffer, sizeof(buffer))))
+      while ((s = string_nextinlist(&list, &sep, NULL, 0)))
 	{
 	int len;
 
-- 
2.1.4

